prov/gni/Makefile.am

```diff
@@ -0,0 +1,123 @@
+
+#
+# want to keep ccan in the include path seen in the
+# source code, so just add prov/gni to the include path
+# rather than prov/gni/ccan
+#
+AM_CFLAGS += -I$(top_srcdir)/prov/gni/include -I$(top_srcdir)/prov/gni/common -I$(top_srcdir)/prov/gni
+
+_gni_files = \
+	prov/gni/common/rbtree.c \
+	prov/gni/common/rbtree.h \
+	prov/gni/common/fasthash/fasthash.c \
+	prov/gni/common/fasthash/fasthash.h \
+	prov/gni/src/gnix_atomic.c \
+	prov/gni/src/gnix_av.c \
+	prov/gni/src/gnix_bitmap.c \
+	prov/gni/src/gnix_cm.c \
+	prov/gni/src/gnix_cm_nic.c \
+	prov/gni/src/gnix_cntr.c \
+	prov/gni/src/gnix_cq.c \
+	prov/gni/src/gnix_datagram.c \
+	prov/gni/src/gnix_dom.c \
+	prov/gni/src/gnix_ep.c \
+	prov/gni/src/gnix_eq.c \
+	prov/gni/src/gnix_fabric.c \
+	prov/gni/src/gnix_freelist.c \
+	prov/gni/src/gnix_hashtable.c \
+	prov/gni/src/gnix_init.c \
+	prov/gni/src/gnix_mbox_allocator.c \
+	prov/gni/src/gnix_mr.c \
+	prov/gni/src/gnix_msg.c \
+	prov/gni/src/gnix_nameserver.c \
+	prov/gni/src/gnix_nic.c \
+	prov/gni/src/gnix_queue.c \
+	prov/gni/src/gnix_rma.c \
+	prov/gni/src/gnix_tags.c \
+	prov/gni/src/gnix_util.c \
+	prov/gni/src/gnix_vc.c \
+	prov/gni/src/gnix_wait.c
+
+_gni_headers = \
+	prov/gni/include/fi_ext_gni.h \
+	prov/gni/include/gnix_atomic.h \
+	prov/gni/include/gnix_av.h \
+	prov/gni/include/gnix_bitmap.h \
+	prov/gni/include/gnix_cm_nic.h \
+	prov/gni/include/gnix_cntr.h \
+	prov/gni/include/gnix_cq.h \
+	prov/gni/include/gnix_datagram.h \
+	prov/gni/include/gnix_ep.h \
+	prov/gni/include/gnix_ep_msg.h \
+	prov/gni/include/gnix_ep_rdm.h \
+	prov/gni/include/gnix_eq.h \
+	prov/gni/include/gnix_freelist.h \
+	prov/gni/include/gnix.h \
+	prov/gni/include/gnix_hashtable.h \
+	prov/gni/include/gnix_mbox_allocator.h \
+	prov/gni/include/gnix_mr.h \
+	prov/gni/include/gnix_msg.h \
+	prov/gni/include/gnix_nameserver.h \
+	prov/gni/include/gnix_nic.h \
+	prov/gni/include/gnix_priv.h \
+	prov/gni/include/gnix_queue.h \
+	prov/gni/include/gnix_rma.h \
+	prov/gni/include/gnix_tags.h \
+	prov/gni/include/gnix_util.h \
+	prov/gni/include/gnix_vc.h \
+	prov/gni/include/gnix_wait.h
+
+if HAVE_CRITERION
+bin_PROGRAMS += prov/gni/test/gnitest
+dist_bin_SCRIPTS = prov/gni/test/run_gnitest
+prov_gni_test_gnitest_SOURCES = \
+	prov/gni/test/allocator.c \
+	prov/gni/test/av.c \
+	prov/gni/test/bitmap.c \
+	prov/gni/test/cancel.c \
+	prov/gni/test/cntr.c \
+	prov/gni/test/cq.c \
+	prov/gni/test/datagram.c \
+	prov/gni/test/dlist-utils.c \
+	prov/gni/test/dom.c \
+	prov/gni/test/ep.c \
+	prov/gni/test/eq.c \
+	prov/gni/test/freelist.c \
+	prov/gni/test/hashtable.c \
+	prov/gni/test/mr.c \
+	prov/gni/test/nic.c \
+	prov/gni/test/pmi_utils.c \
+	prov/gni/test/queue.c \
+	prov/gni/test/rdm_atomic.c \
+	prov/gni/test/rdm_fi_pcd_trecv_msg.c \
+	prov/gni/test/rdm_rma.c \
+	prov/gni/test/rdm_rx_overrun.c \
+	prov/gni/test/rdm_sr.c \
+	prov/gni/test/rdm_tagged_sr.c \
+	prov/gni/test/tags.c \
+	prov/gni/test/utils.c \
+	prov/gni/test/vc.c \
+	prov/gni/test/wait.c
```

Comment:
These are unit tests that typically target gni provider specific interfaces.  They enabled us to get pieces working without the have entire code paths implemented and were very important for our development (and still remain so).  They are not stand alone tests, and rather use a particular testing interface called Criterion (https://github.com/Snaipe/Criterion).